integrationtests/agent/adoption_test.go

```diff
@@ -0,0 +1,387 @@
+				cm.Labels = map[string]string{}
+			})
+			assertBundleDeployment("remove-metadata", bundleDeploymentNotOwnedByUs)
+		})
+	})
+
+	When("a bundle deployment adopts a \"clean\" resource", Label("clean"), func() {
+		It("verifies that the ConfigMap is adopted and its content merged", func() {
+			createConfigMap(map[string]string{"foo": "bar"}, nil, nil)
+			createBundleDeployment("adopt-clean", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with wrangler metadata", Label("wrangler-metadata"), func() {
+		It("verifies that the ConfigMap is adopted, its content merged and ownership changed", func() {
+			const (
+				objectSetHashKey   = "objectset.rio.cattle.io/hash"
+				objectSetIDKey     = "objectset.rio.cattle.io/id"
+				objectSetHashValue = "33ed67317c57ea78702e369c4c025f8df88553cc"
+				objectSetIDValue   = "some-assumed-old-id"
+			)
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{objectSetHashKey: objectSetHashValue},
+				map[string]string{objectSetIDKey: objectSetIDValue},
+			)
+			createBundleDeployment("adopt-wrangler-metadata", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				if err := mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"}); err != nil {
+					return err
+				}
+				if mapPartialMatch(cm.Annotations, map[string]string{objectSetIDKey: objectSetIDValue}) == nil {
+					return fmt.Errorf("ObjectSet ID should have been updated")
+				}
+				if mapPartialMatch(cm.Labels, map[string]string{objectSetHashKey: objectSetHashValue}) == nil {
+					return fmt.Errorf("ObjectSet Hash should have been updated")
+				}
+				return nil
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with invalid wrangler metadata", Label("wrangler-metadata"), func() {
+		It("verifies that the ConfigMap is adopted and its content merged", func() {
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{"objectset.rio.cattle.io/hash": "234"},
+				map[string]string{"objectset.rio.cattle.io/id": "$#@"},
+			)
+			createBundleDeployment("adopt-invalid-wrangler-metadata", true)
+			assertConfigMap(func(cm corev1.ConfigMap) error {
+				if err := isConfigMapAdopted(&cm); err != nil {
+					return err
+				}
+				return mapPartialMatch(cm.Data, map[string]string{"foo": "bar", "key": "value"})
+			})
+		})
+	})
+
+	When("a bundle deployment adopts a resource with random metadata", Label("random-metadata"), func() {
+		It("verifies that the ConfigMap is adopted and its content merged", func() {
+			createConfigMap(
+				map[string]string{"foo": "bar"},
+				map[string]string{"foo": "234"},
+				map[string]string{"bar": "xzy"},
```

Comment:
nit: this may be a tad easier to read as something like:
```suggestion
				map[string]string{"foo": "bar"},
				map[string]string{"foo-label": "234"},
				map[string]string{"bar-annotation": "xzy"},
```

Reply:
I went backwards here. I had the intention to make it simple, but also had the same doubt about readability. So I went back to using the corev1.ConfigMap instead, but setting name and namespace in helper functions.